main.go

```diff
@@ -49,41 +58,50 @@ func main() {
 		log.Fatal("Failed to create client: ", err)
 	}
 
-	sigChan := make(chan os.Signal)
-	signal.Notify(sigChan, syscall.SIGINT, syscall.SIGTERM)
+	events := make(chan job.Event)
+	go func() {
+		var lastPhase core.PodPhase
+		for event := range events {
+			switch event := event.(type) {
+			case error:
+				log.Printf("Error: %v", event)
+			case job.LogLine:
+				log.Printf("%s: %s", event.Container, event.Line)
+			case core.PodStatus:
+				status := event
+				if status.Phase != lastPhase {
+					log.Print("Phase: ", status.Phase)
+				}
+				lastPhase = status.Phase
+
+				if status.Phase == core.PodPending {
```

Comment:
Minor: You could reverse the logic and `continue` if `status.Phase != core.PodPending`. Saves some nesting.